industrial_ci/src/tests/source_tests.sh

```diff
@@ -97,6 +97,96 @@ function run_clang_tidy_check {
     fi
 }
 
+function ici_combine_cpp_reports {
+  ici_install_pkgs_for_command lcov lcov
+  local target_ws=$1
+  cd "$target_ws" || return 1
+  # Capture initial coverage info
+  lcov --capture --initial \
+       --directory build \
+       --output-file initial_coverage.info | grep -ve "^Processing"
+  # Capture tested coverage info
+  lcov --capture \
+       --directory build \
+       --output-file test_coverage.info | grep -ve "^Processing"
+  # Combine two report (exit function  when none of the records are valid)
+  lcov --add-tracefile initial_coverage.info \
+       --add-tracefile test_coverage.info \
+       --output-file coverage.info || return 0 \
+    && rm initial_coverage.info test_coverage.info
+  # Extract repository files
+  lcov --extract coverage.info "$(pwd)/src/$TARGET_REPO_NAME/*" \
+       --output-file coverage.info | grep -ve "^Extracting"
+  # Filter out test files
+  lcov --remove coverage.info "*/test/*" \
+       --output-file coverage.info | grep -ve "^removing"
+  lcov --list coverage.info
+}
+
+function ici_combine_python_reports {
+  ici_install_pkgs_for_command coverage python3-coverage python3-dev python3-wheel
+  local target_ws=$1
+  cd "$target_ws" || return 1
+  # Find all .coverage file
+  IFS=" " read -r -a python_reports <<< \
+    "$(find "$target_ws/build" \
+             -type f \
+             -name ".coverage" \
+             -printf "%p ")"
+  # Copy coverage file into workspace and
+  # convert names from .coverage to .coverage.0/1/2
+  local arraylength=${#python_reports[@]}
+  if [ "$arraylength" != "0" ]; then
+    for (( i=1; i<arraylength+1; i++ ));
+    do
+      cp "${python_reports[$i-1]}" .coverage."$i"
+    done
+    # Combine coverage files
+    python3 -m coverage combine
+    # Generate report
+    printf "[report]\nomit = \n\t*/test/*\n\t*/setup.py" > .default.coveragerc
```

Comment:
A kind of hackish way to specify files to be ignored for python and coveralls 
(since I cannot do it in run time and coveralls only support **.coveragerc** file for ignore)